planner/core/rule_partition_processor.go

```diff
@@ -730,7 +730,7 @@ func intersectionRange(start, end, newStart, newEnd int) (int, int) {
 }
 
 func (s *partitionProcessor) pruneRangePartition(ctx sessionctx.Context, pi *model.PartitionInfo, tbl table.PartitionedTable, conds []expression.Expression,
-	columns []*expression.Column, names types.NameSlice) (partitionRangeOR, error) {
+	columns []*expression.Column, names types.NameSlice, condsToBePruned *[]expression.Expression) (partitionRangeOR, error) {
```

Comment:
I suggest that we add `condsToBePruned` to the return value instead of using a pointer, which is more go-style:
```
func (s) pruneRangePartition(..., condsToBePruned) (partitionRangeOr, prunedConds, error)
```
Others LGTM.

Reply:
OK

Reply:
After reviewing the source code, I find using pointer might be the proper way right now. Since we also call `pruneRangePartition` function [here](https://github.com/pingcap/tidb/blob/master/planner/core/partition_prune.go#L40), we cannot pass into any information about `condsTobePruned` but nil pointer.

Reply:
In this case, we can use `rangeOr, _, err  = s.pruneRangePartition(..., nil)`.

Reply:
OK